docs/rfcs/751-write-empty-file-behavior.md

```diff
@@ -0,0 +1,51 @@
+- Author:  Abyss-w <mad.hatter@foxmail.com>
+- Start Date: 2021-09-10
+- RFC PR: [beyondstorage/go-storage#751](https://github.com/beyondstorage/go-storage/pull/751)
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+
+# GSP-751: Write Empty File Behavior
+
+## Background
+
+In our definition, the `Write` API will upload a file to the target object. We do not do anything about uploading empty files. However, some services do not support uploading empty files. As [proposed](https://forum.beyondstorage.io/t/topic/204) in our community, for s3, when we upload an empty file, the service will hang permanently. Here is an example of s3:
+
+```go
+_, err = fs.s.Write(path,nil, 0)

```

Comment:
Maybe `fs` should be `s3` to keep consistent with the description above?

Reply:
Yes. It's better this way.